### PR TITLE
MNT Support numpy2

### LINKED ISSUE #425: Support for Numpy 2.0 and higher
`numpy` version is not defined within `pyproject.toml` therefore installing `skops` in an environment with `numpy>=2` raises the following error:

```
AttributeError: `np.sctypes` was removed in the NumPy 2.0 release. Access dtypes explicitly instead.
```

### PR DESCRIPTION
**Pull request by adrinjalali**
fixes #425

This adds support for numpy2. In the process, I also found some other issues which I fixed. Some of them were in tests for model cards, and I think the changed code is nicer since it leaves the titles of the sections unchanged.

### CODE REVIEW

**Comment by adrinjalali** [comment]
CI is green. Except that a single line is not tested since we're testing a recent numpy<2 here.

I'll have another PR to test minimum dependencies in CI, but for now this fixes urgent issues and I think we should merge and release.

cc @BenjaminBossan 

(FYI, numpy issues on the skops side @TamaraAtanasoska)

**Review by BenjaminBossan — APPROVED** [review]
Thanks for updating skops to work with numpy v2 and for adding the other improvements.

File: skops/io/_numpy.py [hunk 1]
```diff
@@ -176,11 +176,12 @@ def _construct(self):
 
 def random_generator_get_state(obj: Any, save_context: SaveContext) -> dict[str, Any]:
     bit_generator_state = get_state(obj.bit_generator.state, save_context)
+    seed_seq_state = get_state(obj.bit_generator.seed_seq.state, save_context)
     res = {
         "__class__": obj.__class__.__name__,
         "__module__": get_module(type(obj)),
         "__loader__": "RandomGeneratorNode",
-        "content": {"bit_generator": bit_generator_state},
+        "content": {"bit_generator": bit_generator_state, "seed_seq": seed_seq_state},
```
[the remark this answers is not in the document]
Comment (adrinjalali) [reply skops/io/_numpy.py:184]: So in `numpy~=1` we have:

```py
>>> np.random.default_rng().__reduce__()
(<function __generator_ctor at 0x777841d798a0>, ('PCG64', <function __bit_generator_ctor at 0x777841d3cc20>), {'bit_generator': 'PCG64', 'state': {'state': 220935942547292961595576198891696034818, 'inc': 281084058663618135801235667263986488477}, 'has_uint32': 0, 'uinteger': 0})
```

while `numpy=2` does this:

```py
>>> np.random.default_rng().__reduce__()
(<function __generator_ctor at 0x778c98b64b80>, (<numpy.random._pcg64.PCG64 object at 0x778c99543480>,), None)
```

and that means our check fails, and while fixing the check going recursively in what's in the output of `__reduce__`, I noticed we're not saving the seed sequence's state. It doesn't seem to have an affect in the next few generated numbers as I checked, but it's something which can be stored and loaded, so I added it.

File: skops/io/tests/_utils.py [hunk 2]
```diff
@@ -113,16 +113,16 @@ def _assert_vals_equal(val1, val2):
...
+        _assert_tuples_equal(val1, val2)
     elif isinstance(val1, float) and np.isnan(val1):
         assert np.isnan(val2)
     elif isinstance(val1, dict):
         # dictionaries are compared by comparing their values recursively.
         assert set(val1.keys()) == set(val2.keys())
         for key in val1:
             _assert_vals_equal(val1[key], val2[key])
+    # elif isinstance(val1, np.random.Generator):
+    #     assert val1.bit_generator.state == val2.bit_generator.state
```
[the remark this answers is not in the document]
Comment (adrinjalali) [reply skops/io/tests/_utils.py:125]: oops, this was my workaround which we don't need anymore. I was being lazy at first since I was totally lost.
